hermes/probe/create_file.go

```diff
@@ -0,0 +1,114 @@
+package probe
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+	"github.com/googleinterns/step224-2020/blob/evan/hermesMain/cmd"
+)
+
+type RandomFile struct {
+	Seed int64
+	Size int // File size in bytes
+}
+
+func (f *RandomFile) NewReader() *randomFileReader {
+	return &randomFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.Seed))}
+}
+
+type randomFileReader struct {
+	size int
+	// Current reading index.
+	i    int
+	rand *rand.Rand
+}
+
+func (r *randomFileReader) Read(b []byte) (n int, err error) {
+	if r.i >= r.size {
+		return 0, io.EOF
+	}
+	if len(b) > r.size-r.i {
+		n, err = r.rand.Read(b[:r.size-r.i])
+	} else {
+		n, err = r.rand.Read(b)
+	}
+	r.i += n
+	return
+}
+
+func (f *RandomFile) generateChecksum() (string, error) {
+	r := f.NewReader()
+	h := sha1.New()
+	buff := make([]byte, 8)
+	for {
+		n, err := r.Read(buff)
+		if err == io.EOF {
+			break
+		}
+		if _, err = io.Copy(h, r); err != nil {
+			return nil, fmt.Errorf("io.Copy: %v", err)
+		}
+
+	}
+
+	// returns checksum in hex notation
+	return fmt.Sprintf("%x", h.Sum(nil)), nil
+}
+
+func (f *RandomFile) generateFileName(fileID int) (string, error) {
+	checksum, err := f.generateChecksum()
+	if err != nil {
+		return nil, err
+	}
+	return fmt.Sprintf("Hermes_%02d_%v", fileID, checksum), nil
+}
+
+func (p *MonitorProbe) CreateFile(ctx context.Context, target string, bucketName string, fileID int, fileSize int) error {
```

Comment:
-[] Since CreateFile doesn't return the file its created, You could add a description at here, at the top explaining where it's stored. 

-[] Also, since this is an exported method, add a comment at the top where you explain what each of your parameters/arguments are and any quirks they might have. Similarly add a section for their returns where you explain every error message or success scenario would look like and what they represent. (just like you did in file_gen.go that is in the master branch)

Reply:
doc strings added